parpack.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef __PARPACK_HPP__
+#define __PARPACK_HPP__
+
+/*
+ * From C++, arpack does not exist.
+ * Arpack is Fortran. ISO_C_BINDING is a gateway from Fortran to C, not C++. But, C++ can "get back" to C.
+ * This is why you find C types in the arpack.hpp (C++ header) to be as safe as possible.
+ */
+
+/*
+ * IMPORTANT: MPI communicators MUST be passed from C to Fortran using MPI_Comm_c2f.
+ *            MPI_Fint MCW = MPI_Comm_c2f(MPI_COMM_WORLD);
+ */
+
+#include "mpi.h"
+
+extern "C" void pssaupd_c(MPI_Fint comm, int & ido, const char * bmat, int n, const char * which, int nev,
```

Comment:
maybe have a single header and just do
```
#ifdef __cplusplus
extern "C" {
#endif
  // blabla

#ifdef __cplusplus
}
#endif
```
? Maintaining two files with identical definition is probably not ideal

Reply:
Sure. For now I was fighting (!) for having a very first version that works ! :D 

When this is merged. I guess it can be a good thing to do what you propose for both arpack (.h, .hpp) and parpack (.h and .hpp)

Reply:
i just wanted to make sure that this happens before the release, otherwise you won't be able to change it in a backward-compatible way.

Reply:
I understand. This can be done quick and fast. I'll PR that after this PR is merged.

@sylvestre 
Is this PR OK for you ? (as it is now)

@davydden 
Just to make sure we are OK: you propose to kill *arpack.hpp and to have only *arpack.h that could be included both from C or C++ ? Right ? I am fine with that.
For now, the only thing I fighted for was really to get working !
The other option would be to include *arpack.h in *arpack.hpp (not tested yet but I guess this could work) and may be more "C++ oriented" (I am fine with that too - this is only a question of style, so it's no big deal to me - to me, what's important is that it works ! :D)

Reply:
More or less... I wish you had some docs or tutorial to explain how to use it.
but I won't block on it.
Please let me know when this can be merged!

Reply:
I've added a README as a doc/example (like I did for arpack). For me you can merge this.

I'll try to PR quickly what @davydden  proposes this afternoon or tomorrow if I can access the web.

Reply:
> you propose to kill *arpack.hpp and to have only *arpack.h that could be included both from C or C++ ? 

correct.

Reply:
It isn't what I call documentation but sure :p

Reply:
@fghoussen if you create a PR, it would be easier to see

Reply:
> It isn't what I call documentation but sure :p

Documenting arpack is "following the examples" !... :D 
For icb, this is the only one way to go (like hdf5) : it's commented in the .c and .cpp files (the README redirect to them).

Reply:
> For icb, this is the only one way to go

[doxygen](http://www.stack.nl/~dimitri/doxygen/)?

Reply:
>doxygen ?

Too much of a burden ! I am really running out of time for this... Can't afford that. But I let you do that if you feel this is necessary

>@fghoussen if you create a PR, it would be easier to see

PR'ed !

Reply:
> Too much of a burden !

fully agree, i just wanted to mention that other options.

> But I let you do that if you feel this is necessary

certainly not necessary enough to commit to it ;-)